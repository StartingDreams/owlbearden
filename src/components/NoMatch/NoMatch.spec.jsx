import React from 'react';
import { shallow } from 'enzyme';
import { fromJS } from 'immutable';
import toJson from 'enzyme-to-json';
import { NoMatch, mapStateToProps } from './NoMatch';

describe('<NoMatch />', () => {
  it('mapStateToProps - isAuthenticated should be passed in', () => {
    const state = {
      auth: fromJS({ isAuthenticated: true }),
    };
    const props = mapStateToProps(state);
    expect(props.isAuthenticated).toBe(true);
  });
  it('authenticated users should match the dashboard snapshot', () => {
    const wrapper = shallow(<NoMatch isAuthenticated />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it('unauthenticated users should match the login snapshot', () => {
    const wrapper = shallow(<NoMatch isAuthenticated={false} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
