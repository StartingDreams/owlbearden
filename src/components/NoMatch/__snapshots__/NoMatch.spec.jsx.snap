// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NoMatch /> authenticated users should match the dashboard snapshot 1`] = `
<Redirect
  push={false}
  to={
    Object {
      "pathname": "/dashboard",
    }
  }
/>
`;

exports[`<NoMatch /> unauthenticated users should match the login snapshot 1`] = `
<Redirect
  push={false}
  to={
    Object {
      "pathname": "/login",
    }
  }
/>
`;
